src/client/components/Fetcher/Fetcher.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+
+const baseUrl = '';
+
+async function getUserData(search) {
+  const response = await fetch(`${baseUrl}${encodeURIComponent(search)}`);
+  const result = await response.json();
+  return result;
+}
+
+// eslint-disable-next-line react/prop-types
+export const Fetcher = ({ search, render }) => {
```

Comment:
This whole component can be refactored into a hook. Or you can leave it like this.